web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
+    this.initMap();
+  }
+
+  ngOnChanges() {
+    this.initMap();
+  }
+
+  initMap() {
     this.subscription.add(
       combineLatest([
         this.activeSurvey$,
         this.lois$,
         this.navigationService.getLocationOfInterestId$(),
       ]).subscribe(([survey, lois, locationOfInterestId]) => {
         const loisMap = ImmutableMap(
-          (this.showPredefinedLoisOnly
-            ? lois.filter(loi => loi.predefined !== false)
-            : lois
-          ).map(loi => [loi.id, loi])
+          lois
+            .filter(loi =>
+              this.showPredefinedLoisOnly ? loi.predefined !== false : loi
+            )
+            .filter(loi =>
+              this.selectedJobId !== '' ? loi.jobId === this.selectedJobId : loi
```

Comment:
Same question here. Can this be rewritten without evaluating the truthiness of `loi`?

Reply:
Done.